### PR TITLE
Highlight in Diagram Search

### PR DESCRIPTION
**Pull request by Cyclonit**
Follow up to #1244.

1. Add search button to the TopLeftPanel.
2. Add highlighting of matched keywords to diagram search.

GIF 24-11-2024 07-29-19

### CODE REVIEW

File: packages/diagram/src/LikeC4Search.tsx [hunk 1]
```diff
@@ -1,83 +1,142 @@
...
+      leftSection: <IconSitemap />
+    }));
   }
 
-  const actions: SpotlightActionGroupData[] = useMemo(() => [
-    getNodeActions(),
-    getViewActions()
-  ], [model, store, view])
+  const nodeActions = buildFilteredActions(useMemo(getNodeActionsData, [model, store, view]), query);
+  const viewActions = buildFilteredActions(useMemo(getViewActionsData, [model, store, view]), query);
```
Comment (davydkov) [inline packages/diagram/src/LikeC4Search.tsx]: What about wrapping in single `useMemo`, like:

```suggestion
  const { nodeActions, viewActions } = useMemo(() => ({
      nodeActions: buildFilteredActions(getNodeActionsData, query),
      viewActions: buildFilteredActions(getViewActionsData, query)
    }),
    [model, store, view, query]
  )
```

File: packages/diagram/src/ui/top-left/TopLeftPanel.tsx [hunk 2]
```diff
@@ -174,6 +176,13 @@ export const TopLeftPanel = () => {
         gap={'xs'}>
         {showNavigationButtons && <BackwardForwardButtons />}
         <ActionIconGroup className={css.actionIconGroup} orientation="vertical">
+          <Tooltip label="Search">
```
Comment (davydkov) [inline packages/diagram/src/ui/top-left/TopLeftPanel.tsx]: I think we need to pass `enableSearch` from props to store, like:

https://github.com/likec4/likec4/blob/3c350b44f3602eaa3aaede2d57ff0e725b3d5c22/packages/diagram/src/LikeC4Diagram.tsx#L139

And show this icon if flag is set.

**Review by davydkov — CHANGES_REQUESTED** [review]
Hey @Cyclonit, great job!
Congratulations on your first contribution.

I would ask you to add `enableSearch` to store, add to `DiagramInitialState` and update `LikeC4Diagram`

https://github.com/likec4/likec4/blob/3c350b44f3602eaa3aaede2d57ff0e725b3d5c22/packages/diagram/src/LikeC4Diagram.tsx#L115-L118

I think we can remove prop from `LikeC4DiagramInnerMemo`, as we can read from store

File: packages/diagram/src/LikeC4Search.tsx (hunk 1 above)
Comment (Cyclonit) [reply packages/diagram/src/LikeC4Search.tsx]: Changed as requested.

File: packages/diagram/src/ui/top-left/TopLeftPanel.tsx (hunk 2 above)
Comment (Cyclonit) [reply packages/diagram/src/ui/top-left/TopLeftPanel.tsx]: I think I did what you wanted. I am not entirely certain whether I looked in the right places though.

**Review by davydkov — APPROVED** [review]

**Pull request merged**